spec/02-integration/05-proxy/33-request-id_spec.lua

```diff
@@ -0,0 +1,200 @@
+local utils = require "kong.tools.utils"
+local helpers = require "spec.helpers"
+local pl_file = require "pl.file"
+
+local to_hex = require "resty.string".to_hex
+
+local TEST_CONF = helpers.test_conf
+
+
+local function get_request_id_from_logs(log_pattern)
+  local request_id
+  log_pattern = log_pattern or ""
+  assert
+  .eventually(function()
+    local logs = pl_file.read(TEST_CONF.prefix .. "/" .. TEST_CONF.proxy_error_log)
+    if not logs then
+      return false
+    end
+
+    local _
+    _, _, request_id = logs:find(log_pattern .. ".-, request_id: \"(%x+)\"")
+    return request_id ~= nil
+  end)
+  .with_timeout(5)
+  .ignore_exceptions(true)
+  .is_truthy()
+  return request_id
+end
+
+
+for _, strategy in helpers.each_strategy() do
+describe("Request ID error log tests #" .. strategy, function()
+  local service
+  local proxy_client
+  local log_pattern = "hello"
+  local error_pattern = "error%-generator"
+
+  lazy_setup(function()
+    local bp = helpers.get_db_utils(strategy, { "services", "routes", "plugins" }, { "error-generator" })
+
+    service = bp.services:insert()
+
+    bp.routes:insert({
+      service = service,
+      hosts = { "no_plugins" },
+    })
+
+    local otel_route = bp.routes:insert({
+      service = service,
+      hosts = { "otel_host" },
+    })
+
+    local correlation_route = bp.routes:insert({
+      service = service,
+      hosts = { "correlation_host" },
+    })
+
+    local otel_correlation_route = bp.routes:insert({
+      service = service,
+      hosts = { "otel_correlation_host" },
+    })
+
+    local runtime_error_route = bp.routes:insert({
+      service = service,
+      hosts = { "runtime_error_host" },
+    })
+
+    bp.plugins:insert({
+      name = "opentelemetry",
+      route = { id = otel_route.id },
+      config = {
+        endpoint = "http://localhost:8080/v1/traces",
+      }
+    })
+
+    bp.plugins:insert({
+      name = "opentelemetry",
+      route = { id = otel_correlation_route.id },
+      config = {
+        endpoint = "http://localhost:8080/v1/traces",
+      }
+    })
+
```

Comment:
Since `zipkin` is affected by this change, consider adding a testcase for it

Tests should cover the following combinations


1. Zipkin, OTEL, correlation-id
2. Zipkin, correlation-id
3. OTEL, correlation-id
4. OTEL
5. ZIPKIN
6. correlation
7. None